### PR TITLE
add class method "increment"

### LINKED ISSUE #6359: [Suggestion] Increment fields using Model.
We can increment fields by using [instance.increment(fields, [options])](http://docs.sequelizejs.com/en/latest/api/instance/#incrementfields-options-promisethis).

But I think increment with Model is also necessary. Like:

``` js
  const fields = { answer: 42, tries: 1};
  someModel.increment(fields, {
     where: {
      id: 111,
    }
  });
```

Thanks.

### PR DESCRIPTION
**Pull request by hieven**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [X] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [X] Does your issue contain a link to existing issue (Closes #[issue]) or a description of the issue you are solving?
- [X] Have you added new tests to prevent regressions?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [ ] Have you added an entry under `Future` in the changelog?

_NOTE: these things are not required to open a PR and can be done afterwards / while the PR is open._

### Description of change

I move the instance method `increment` code to be class method `increment`.
The original instance method `increment` will simply add `this.where()` to `options.where` and then call `this.constructor.increment()`.

This is the very first open source contribution I do.
Please give me any advices to improve this PR :-)

I will write document and changelog if this PR is acceptable

close #6359

### CODE REVIEW

**Comment by hieven** [comment]
the test fails in a strange line.
If someone could help figure this out, it would be great

File: lib/query-interface.js [hunk 1]
```diff
@@ -641,18 +641,18 @@ class QueryInterface {
...
+  increment(model, tableName, values, identifier, options) {
+    options = Utils.cloneDeep(options);
 
-    options = _.clone(options) || {};
+    const sql = this.QueryGenerator.arithmeticQuery('+', tableName, values, identifier, options, options.attributes);
+
+    options.model = model;
 
-    options.type = QueryTypes.UPDATE;
-    options.instance = instance;
```
Comment (janmeier) [inline lib/query-interface.js]: This line is what makes `returning: true` work - It shouldn't be removed :)

File: test/unit/increment.test.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            dataValues: {id: 3}
+          })
+        );
+      });
+
+      after(function() {
+        stub.restore();
+      });
+
+      it('should allow increments even if options are not given', function () {
```
Comment (janmeier) [inline test/unit/increment.test.js]: We should require `where` to be defined, like we do for update, so users don't accidentally update all rows in the database

File: lib/model.js [hunk 3]
```diff
@@ -2837,6 +2837,90 @@ class Model {
...
+    for (const attr of Object.keys(values)) {
+      // Field name mapping
+      if (this.rawAttributes[attr] && this.rawAttributes[attr].field && this.rawAttributes[attr].field !== attr) {
+        values[this.rawAttributes[attr].field] = values[attr];
+        delete values[attr];
+      }
+    }
+
+    if (!options.increment) {
+      return this.sequelize.getQueryInterface().decrement(this, this.getTableName(options), values, where, options).then(affectedRows => {
```
Comment (janmeier) [inline lib/model.js]: The `then` handlers are duplicated here

**Comment by hieven** [comment]
Thanks for the quick help and the test finally passes!

I'd like to receive more suggestions for improving this PR since this is the very first contribution I make for open source. :-)

**Comment by janmeier** [comment]
@hieven I don't have any other comments I think, code style looks good, and I like your thoroughness in testing :)

**Comment by hieven** [comment]
@janmeier 
Thank you for the comment. :-)

**Pull request merged**